File: src/contestatal01-py/coins.py
def greedy(n,s):
    coins = []
    for i in range(1,n+1):
        coins.append(i)
    sum = 0
    coinIndex = len(coins)-1
    numOfCoins = 0
    while(sum < s):
        nextCoin = coins[coinIndex]
        if(sum + nextCoin <= s):
            sum += nextCoin
            numOfCoins += 1
        else:
            diff = s-sum 
            nextCoin = coins[diff-1]
            sum += nextCoin 
            numOfCoins += 1
    return numOfCoins     

if __name__ == "__main__":
    inputStr = input ("")
    input = inputStr.split(" ")
    answer = greedy (int(input[0]), int(input[1]))
    print(answer)

File: src/Coins.java
import java.util.Arrays;
import java.util.Scanner;

public class Coins {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String inputStr = sc.nextLine();
        String[] numbersAsStr = inputStr.split(" ");
        long[] input = Arrays.stream(numbersAsStr)
                .mapToLong(Long::parseLong)
                .toArray();
        int answer = greedy(input[0],input[1]);
        System.out.println(answer);
    }

    private static int greedy(long n,long s) {
        int[] coins = new int[(int)n];
        for(int i =1; i <= n; i++){
            coins[i-1]=i;
        }
        long sum = 0;
        int coinIndex = coins.length-1;
        int numOfCoins = 0;

        while(sum < s){
            int coin = select(coins,coinIndex,sum,s);
            sum += coin;
            numOfCoins++;
        }

        return numOfCoins;
    }

    private static int select(int[] coins, int i ,long partialSum, long s){
        int maxCoin = coins[i];
        if(partialSum + maxCoin <= s){
            return maxCoin;
        } else {
            return select(coins,i-1,partialSum,s);
        }
    }
}
